### PR TITLE
Wrote RejectCommandTest and RejectCommandParserTest

### LINKED ISSUE #57: Create test cases for RejectCommand
To ensure that `RejectCommand` works as intended by the team's business logic

### LINKED ISSUE #58: Create test cases for RejectCommandParser
To ensure that RejectCommandParser works as intended by the team's business logic

### PR DESCRIPTION
**Pull request by FredericChow00**
## What
Add test cases for `RejectCommand`, `RejectCommandParserTest` and some other related Test classes.

## Why
Insufficient test cases for the above mentioned classes and also changes were made to other Test classes to enhance code quality.

This ensures that the codebase is operating according to the intended business logic.

## How
Added test cases for public methods in the `RejectCommand` and `RejectCommandParser` class.

Fix #57
Fix #58

### CODE REVIEW

File: src/test/java/seedu/address/logic/parser/RejectCommandParserTest.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
+//package seedu.address.logic.parser;
```
Comment (FredericChow00) [inline src/test/java/seedu/address/logic/parser/RejectCommandParserTest.java:1]: To be continued in another PR :)

File: src/test/java/seedu/address/logic/commands/RejectCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+
+        String expectedMessage = String.format(RejectCommand.MESSAGE_REJECT_PERSON_SUCCESS, rejectedPerson.getName());
+
+        expectedModel.setPerson(personToReject, rejectedPerson);
+        expectedModel.updateFilteredPersonList(namePhonePredicate);
+
+        assertCommandSuccess(rejectCommand, model, expectedMessage, expectedModel);
+    }
+
+    /**
```
Comment (Merrickneo) [inline src/test/java/seedu/address/logic/commands/RejectCommandTest.java:92]: Could change ~Edit~ to **Edits** for greater clarity

Comment (Merrickneo) [reply src/test/java/seedu/address/logic/commands/RejectCommandTest.java:92]: Could split up the comment into different statements to talk about the 2 things you are testing in this test case.
1. Filters the list by phone number
2. Checks if user is able to reject applicants that are not shown on the current filteredList

File: src/test/java/seedu/address/model/tag/TagTest.java [hunk 3]
```diff
@@ -0,0 +1 @@
+
```
Comment (Merrickneo) [inline src/test/java/seedu/address/model/tag/TagTest.java:1]: Remove this file/directory as we are not using `Tag` anymore

File: src/test/java/seedu/address/testutil/TypicalPersons.java [hunk 4]
```diff
@@ -38,7 +38,7 @@ public class TypicalPersons {
             .withNotes("owesMoney", "friends").build();
     public static final Person CARL = new PersonBuilder().withName("Carl Kurz").withPhone("95352563")
             .withEmail("heinz@example.com").withAddress("wall street")
-            .withStatus("ACCEPTED").withInterviewDateTime("").build();
+            .withStatus("ACCEPTED").withInterviewDateTime("04-03-2024 13:00").build();
```
Comment (Merrickneo) [inline src/test/java/seedu/address/testutil/TypicalPersons.java:41]: Could have a constant variable defined and made use of here to enable easier code mantainance in the future

**Review by Merrickneo — DISMISSED** [review]
Could remove the `Tag` test as we are not using `Tag` anymore.

Also, could make use of more constants to help with testing!

Good work on the testing!

File: src/test/java/seedu/address/logic/commands/RejectCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+
+        String expectedMessage = String.format(RejectCommand.MESSAGE_PERSON_CANNOT_BE_REJECTED,
+                rejectedPerson.getName());
+
+        assertCommandFailure(rejectCommand, model, expectedMessage);
+    }
+
+    @Test
+    public void execute_invalidNameUnfilteredList_failure() {
+        Person personToReject = model.getFilteredPersonList().get(INDEX_THIRD_PERSON.getZeroBased());
```
Comment (bofeng1999) [inline src/test/java/seedu/address/logic/commands/RejectCommandTest.java:136]: Maybe can make use of `NamePhoneNUmberPredicate`  instead?

**Review by bofeng1999 — APPROVED** [review]
LGTM, just a minor comments from my side!

File: src/test/java/seedu/address/logic/commands/RejectCommandTest.java (hunk 2 above)
Comment (FredericChow00) [reply src/test/java/seedu/address/logic/commands/RejectCommandTest.java:92]: Ok will do! Thanks for the feedback!

File: src/test/java/seedu/address/logic/commands/RejectCommandTest.java (hunk 5 above)
Comment (FredericChow00) [reply src/test/java/seedu/address/logic/commands/RejectCommandTest.java:137]: Yup I used the `NamePhoneNumberPredicate` in `line 140`. I just wanted to get the person in the 3rd index (as an example) then use the person's name and phone number to create the `NamePhoneNumberPredicate` in `line 140`.

File: src/test/java/seedu/address/testutil/TypicalPersons.java (hunk 4 above)
Comment (FredericChow00) [reply src/test/java/seedu/address/testutil/TypicalPersons.java]: Updated!

File: src/test/java/seedu/address/model/tag/TagTest.java (hunk 3 above)
Comment (FredericChow00) [reply src/test/java/seedu/address/model/tag/TagTest.java]: Removed!

**Review by FredericChow00 — COMMENTED** [review]
Changed according to feedback given
